en: add doc for deploy across eks/gke (#1593)
pingcap/docs-tidb-operator: en/build-multi-gcp-gke.md

    ```bash
    gcloud compute networks create ${network_name} --subnet-mode=custom
    ```

2. In the VPC network created above, create three subnets that belong to different regions. The CIDR block of each subnet does not overlap with that of each other.

    {{< copyable "shell-regular" >}}

    ```bash
    gcloud compute networks subnets create ${subnet_1} \
        --region=${region_1} \
        --network=${network_name} \
        --range=10.0.0.0/16 \
        --secondary-range pods=10.10.0.0/16,services=10.100.0.0/16
    ```

    {{< copyable "shell-regular" >}}

    ```bash
    gcloud compute networks subnets create ${subnet_2} \
        --region=${region_2} \
        --network=${network_name} \
        --range=10.1.0.0/16 \
        --secondary-range pods=10.11.0.0/16,services=10.101.0.0/16
    ```

    {{< copyable "shell-regular" >}}

    ```bash
    gcloud compute networks subnets create ${subnet_3} \
        --region=${region_3} \
        --network=${network_name} \
        --range=10.2.0.0/16 \
        --secondary-range pods=10.12.0.0/16,services=10.102.0.0/16
    ```

    `${subnet_1}`, `${subnet_2}`, and `${subnet_3}` refer to the names of the three subnets.

>     `--range=10.0.0.0/16` specifies the CIDR block of the subnet in the cluster. The CIDR blocks of all cluster subnets **must not** overlap with each other.

Reviewer: ```suggestion
    `--range=10.0.0.0/16` specifies the CIDR block of the `${subnet_1}` in the cluster. The CIDR blocks of all cluster subnets **must not** overlap with each other.
```

    `--secondary-range pods=10.11.0.0/16,services=10.101.0.0/16` specifies the CIRD block used by Kubernetes Pods and Services. This CIRD block will be used later.

## Step 2. Start the Kubernetes cluster

> Create three GKE clusters, and each cluster uses the subnet created in Step 1.

Reviewer: ```suggestion
Create three GKE clusters, and each cluster uses one of the subnets created in Step 1.
```

    *         gke_pingcap_us-west1_tidb-1   gke_pingcap_us-west1_tidb-1   gke_pingcap_us-west1_tidb-1
              gke_pingcap_us-west2_tidb-2   gke_pingcap_us-west2_tidb-2   gke_pingcap_us-west2_tidb-2
              gke_pingcap_us-west3_tidb-3   gke_pingcap_us-west3_tidb-3   gke_pingcap_us-west3_tidb-3
    ```

    In the following sections, `${context_1}`, `${context_2}`, and `${context_3}` refer to the context of each cluster.

### Configure the firewall rules

1. Update the firewall rules for Cluster 1.

    1. Obtain the name of the firewall rule used for communication between GKE Pods. The name of the firewall rule is similar to `gke-${cluster_1}-${hash}-all`.

        {{< copyable "shell-regular" >}}

        ```bash
        gcloud compute firewall-rules list --filter='name~gke-${cluster_1}-.*-all'
        ```

        The expected output is as follows. The rule name is in the `NAME` column.

        ```
        NAME                           NETWORK     DIRECTION  PRIORITY  ALLOW                         DENY  DISABLED
        gke-${cluster_1}-b8b48366-all  ${network}  INGRESS    1000      tcp,udp,icmp,esp,ah,sctp            False
        ```

>     2. Update the source range of the firewall rule. Add the CIDR block of the Pod network of the other two clusters to the source range:

Reviewer: ```suggestion
    2. Update the source range of the firewall rule. Add the CIDR blocks of the Pod network of the other two clusters to the source range:
```

## Step 3. Verify the network interconnectivity

Before you deploy the TiDB cluster, you need to verify that the network between the GKE clusters is interconnected.

1. Save the following content in the `sample-nginx.yaml` file.

    ```yaml
    apiVersion: v1
    kind: Pod
    metadata:
      name: sample-nginx
      labels:
        app: sample-nginx
    spec:
      hostname: sample-nginx
      subdomain: sample-nginx-peer
      containers:
      - image: nginx:1.21.5
        imagePullPolicy: IfNotPresent
        name: nginx
        ports:
          - name: http
            containerPort: 80
      restartPolicy: Always
    ---
    apiVersion: v1
    kind: Service
    metadata:
      name: sample-nginx-peer
    spec:
      ports:
        - port: 80
      selector:
        app: sample-nginx
      clusterIP: None
    ```

2. Deploy the nginx service in the namespaces of three clusters.

    {{< copyable "shell-regular" >}}

    ```bash
    kubectl --context ${context_1} -n default apply -f sample-nginx.yaml

    kubectl --context ${context_2} -n default apply -f sample-nginx.yaml

    kubectl --context ${context_3} -n default apply -f sample-nginx.yaml
    ```

> 3. Access the nginx services of other clusters to verify the network interconnectivity.

Reviewer: ```suggestion
3. Access the nginx services of each cluster to verify the network interconnectivity.
```

    The following command verifies the network from Cluster 1 to Cluster 2:

    {{< copyable "shell-regular" >}}

    ```bash
    kubectl --context ${context_1} exec sample-nginx -- curl http://sample-nginx.sample-nginx-peer.default.svc.${cluster_domain_2}:80
    ```

    If the output is the welcome page of nginx, the network is connected.

4. After the verification, delete the nginx services:

    {{< copyable "shell-regular" >}}

    ```bash

>     kubectl --context ${context_2} -n default delete -f sample-nginx.yaml

Reviewer: ```suggestion
    kubectl --context ${context_1} -n default delete -f sample-nginx.yaml
```
